src/components/ShortenedLink.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { pallete } from "../misc/pallete";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faX } from "@fortawesome/free-solid-svg-icons";
+import Button from "./Button";
+import { motion } from "framer-motion";
+import PropTypes from "prop-types";
+const isLink = function (props, propName, componentName) {
```

Comment:
dobrze jest funkcje wyrzucać do osobnych plików - jeśli jest to funkcja, która ma szansę być użyta w większej ilości miejsc w aplikacji to wtedy tworzy się w folderze 'src' folder 'utils' i często tam się wrzuca takie funkcje, ale w w tym przypadku gdy mamy funkcję, którą użyjemy raz to tworzy się plik w folderze danego komponentu. Nie ma jednego sposobu nazewnictwa tych plików, ja lubię notację  podobną jak wyżej proponowałem do styled componentów, czyli 'NazwaComponentu.utils.js'
czyli tutaj po prostu: ShortenedLink.utils.ts

Reply:
Ma to takie zalezy, że - łatwo się wyszukuje takie pliki - mimo, że nazwa jest skomplikowana to wystarczy, że wiemy w którym komponencie jest dana funkcja i możemy łatwo ją wyszukać szukając po nazwie komponentu. Tworzy nam to też hermetyczne środowisko danego komponentu - w folderze znajduje się wszystko co jest potrzebne, aby taki komponent mógł w pełni funkcjonować, łatwiej się też testuje taki komponent, jeśli mamy wydzielone warty logiki od warstwy styli i warstwy wizualnej.

Reply:
No i też jest w programowaniu (w konkteście programowania obiektowego) wzorzec SOLID: https://pl.wikipedia.org/wiki/SOLID_(programowanie_obiektowe)
w którym litera S oznacza 'single responsibility' - nie do końca można to odwzorować w przypadku Reacta (dotyczy to ciut innej kwestii) ale część tej formuły jest uniwersalna - 'plik powinien mieć jedną odpowiedzialność' - czyli osobny plik na wartstwę wizualną (style), osobny do struktury, osobny do pracy z api, osobny do dodatkowych funkcji (utils) itd